client/src/components/application/SelectBox/index.js

```diff
@@ -12,12 +12,14 @@ export default class SelectBox extends React.Component {
   };
 
   handleChange = (event) => {
+    const { setval } = this.props;
     this.setState({ selectValue: event.target.value });
+    setval(event);
```

Comment:
selectValue here not used any where ?! if its not required just delete it , else you can add it for the function setval instead of adding event a lone